backend/module/chaos/serverexperimentation/rtds/rtds_cache.go

```diff
@@ -186,11 +185,27 @@ func setSnapshot(snapshotCache cacheWrapper, rtdsLayerName string, cluster strin
 	// No experiments meaning clear all experiments for the given upstream cluster
 	if len(experiments) != 0 {
 		for _, experiment := range experiments {
-			testConfig := &serverexperimentation.TestConfig{}
-			if !isFaultTest(experiment, testConfig) {
+			httpFaultConfig := &serverexperimentation.HTTPFaultConfig{}
+			if !isFaultTest(experiment, httpFaultConfig) {
 				continue
 			}
-			percentageKey, percentageValue, faultKey, faultValue := createRuntimeKeys(testConfig, ingressPrefix, egressPrefix, logger)
+
+			upstreamCluster, downstreamCluster, err := unParse(httpFaultConfig)
+			if err != nil {
+				logger.Errorw("Invalid http fault config", "config", httpFaultConfig)
```

Comment:
This and the other error conditions below should probably `continue` or `return`?